src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -44,6 +54,62 @@ def infer_schema_from_description(self, table_or_query: str) -> StructType:
             cursor.close()
             conn.close()
 
+    def udtf_ingestion(
+        self,
+        session: "snowflake.snowpark.Session",
+        schema: StructType,
+        partition_table: str,
+        external_access_integrations: str,
+        fetch_size: int = 1000,
+        imports: Optional[List[str]] = None,
+        packages: Optional[List[str]] = None,
+    ) -> "snowflake.snowpark.DataFrame":
+        from snowflake.snowpark._internal.data_source.utils import UDTF_PACKAGE_MAP
+
+        udtf_name = f"data_source_udtf_{generate_random_alphanumeric(5)}"
+        session.udtf.register(
+            self.udtf_class_builder(fetch_size=fetch_size),
+            name=udtf_name,
+            output_schema=StructType(
+                [
+                    StructField(field.name, StringType(), field.nullable)
```

Comment:
I forgot the context, can you help me remember why for output schema each column uses StringType not VariantType?

Reply:
that is from Yijun's suggestion, I kind of forget why, but looking at snowflake's doc, I think variant might be better